app/code/Magento/InventoryCatalogApi/Api/BulkInventoryTransferInterface.php

```diff
@@ -0,0 +1,26 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\InventoryCatalogApi\Api;
+
+/**
+ * Perform bulk product inventory transfer
+ *
+ * @api
+ */
+interface BulkInventoryTransferInterface
+{
+    /**
+     * Run bulk inventory transfer
+     * @param string[] $skus
+     * @param string $destinationSource
+     * @param bool $defaultSourceOnly
+     * @return bool
+     * @throws \Magento\Framework\Validation\ValidationException
+     */
+    public function execute(array $skus, string $destinationSource, bool $defaultSourceOnly = false): bool;
```

Comment:
usually, optional parameters are `smell` in the code because they lead to conditional logic inside the implementation and thus a violation of polymorphism.

This case is not an exception, moreover, currently your program interface is a representation of limitations you have on UI . 
Because UI does not provide an ability to move Inventory from Source A -> Source B

It provides an ability to move all the Inventory (among all the sources) to specified Source.
Thus, if there are `Source A, Source B, Default Source`
and SKU is assigned to `Source A, Source B` 
it's not possible to transfer just a stock for SKU from Source A to Default Source
all merchant can do is to move all available stock `Source A + Source B` to Default Source

And to bypass this limitation `$defaultSourceOnly` parameter has been introduced, which provides an ability to Migrate data from Default Source only even so the SKU is assigned to other sources as well. 

So, why not to provide an API which gives an ability to migrate data from Source A -> Source B ? in this case we no need any special case like with Default Source.

Does it sound reasonable @phoenix128

Reply:
I wanted to use the same approach as defined in the acceptance criteria to reflect the admin functions, but you are right. In anyway this will be refactored with the new inventory trnasfer discussed in the grooming session.